In [1]:
import pandas as pd

ARCHIVO "HOMICIDIOS.XLSX"

In [38]:
df = pd.read_excel("../PI-2/01 - Datasets/Datasets Siniestros Viales/homicidios.xlsx", sheet_name= "HECHOS")
df.head(2)

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA HH  \
0  2016-0001           1 2016-01-01  2016   1   1  04:00:00  4   
1  2016-0002           1 2016-01-02  2016   1   2  01:15:00  1   

                             LUGAR_DEL_HECHO TIPO_DE_CALLE  ... Altura  \
0  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...    NaN   
1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...    NaN   

                                 Cruce  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                  DE LOS CORRALES AV.   

                               Dirección Normalizada COMUNA  \
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   

                               XY (CABA)         pos x         pos y  \
0  Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1  Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   

    PARTICIPANTES VICTIMA    ACUSADO  
0       MOTO-AUTO    MOTO       AUTO  
1  AUTO-PASAJEROS    AUTO  PASAJEROS  

[2 rows x 21 columns]

In [39]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 696 entries, 0 to 695
Data columns (total 21 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   ID                     696 non-null    object        
 1   N_VICTIMAS             696 non-null    int64         
 2   FECHA                  696 non-null    datetime64[ns]
 3   AAAA                   696 non-null    int64         
 4   MM                     696 non-null    int64         
 5   DD                     696 non-null    int64         
 6   HORA                   696 non-null    object        
 7   HH                     696 non-null    object        
 8   LUGAR_DEL_HECHO        696 non-null    object        
 9   TIPO_DE_CALLE          696 non-null    object        
 10  Calle                  695 non-null    object        
 11  Altura                 129 non-null    float64       
 12  Cruce                  525 non-null    object        
 13  Direc

Solamente las columnas "Altura" y "Cruce" tienen datos nulos, pero eso se debe a que cuando el siniestro ocurre en un cruce de calle, la altura es dato vacío. La sumatoria de dichos valores NO nulos resulta en 654 valores. 

Por ende, de los 696 registros, hay 42 valores que en ambas columnas se encuentran vacios. A explorar rapidamente esos datos en el dataset, se denota que los mismos indican ocurrencia en AUTOPISTAS.

In [40]:
df = df[["ID", "AAAA", "MM", "HH", "FECHA", "TIPO_DE_CALLE", "Calle", "Cruce", "COMUNA", "pos x", "pos y", "VICTIMA", "ACUSADO"]]
df["FECHA"] = df["FECHA"].dt.day_name()
df.head(2)

ID  AAAA  MM HH     FECHA TIPO_DE_CALLE             Calle  \
0  2016-0001  2016   1  4    Friday       AVENIDA  PIEDRA BUENA AV.   
1  2016-0002  2016   1  1  Saturday      GRAL PAZ    PAZ, GRAL. AV.   

                                 Cruce  COMUNA         pos x         pos y  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.       8  -58.47533969  -34.68757022   
1                  DE LOS CORRALES AV.       9  -58.50877521  -34.66977709   

  VICTIMA    ACUSADO  
0    MOTO       AUTO  
1    AUTO  PASAJEROS

In [41]:
df_homicidios = df.rename(columns={"AAAA":"Año", "MM":"Mes", "HH":"Hora", "FECHA":"Dia", "TIPO_DE_CALLE":"Lugar", "Calle":"Calle_ppal", "Cruce":"Calle_sec", "COMUNA": "Comuna", "pos x": "Longitud", "pos y": "Latitud", "VICTIMA": "Victima", "ACUSADO": "Acusado"})
df_homicidios.head(2)

ID   Año  Mes Hora       Dia     Lugar        Calle_ppal  \
0  2016-0001  2016    1    4    Friday   AVENIDA  PIEDRA BUENA AV.   
1  2016-0002  2016    1    1  Saturday  GRAL PAZ    PAZ, GRAL. AV.   

                             Calle_sec  Comuna      Longitud       Latitud  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.       8  -58.47533969  -34.68757022   
1                  DE LOS CORRALES AV.       9  -58.50877521  -34.66977709   

  Victima    Acusado  
0    MOTO       AUTO  
1    AUTO  PASAJEROS

In [42]:
df_victimas = pd.read_excel("../PI-2/01 - Datasets/Datasets Siniestros Viales/homicidios.xlsx", sheet_name= "VICTIMAS")
df_victimas.head(2)

ID_hecho      FECHA  AAAA  MM  DD        ROL VICTIMA       SEXO EDAD  \
0  2016-0001 2016-01-01  2016   1   1  CONDUCTOR    MOTO  MASCULINO   19   
1  2016-0002 2016-01-02  2016   1   2  CONDUCTOR    AUTO  MASCULINO   70   

   FECHA_FALLECIMIENTO  
0  2016-01-01 00:00:00  
1  2016-01-02 00:00:00

In [43]:
df_victimas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717 entries, 0 to 716
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   ID_hecho             717 non-null    object        
 1   FECHA                717 non-null    datetime64[ns]
 2   AAAA                 717 non-null    int64         
 3   MM                   717 non-null    int64         
 4   DD                   717 non-null    int64         
 5   ROL                  717 non-null    object        
 6   VICTIMA              717 non-null    object        
 7   SEXO                 717 non-null    object        
 8   EDAD                 717 non-null    object        
 9   FECHA_FALLECIMIENTO  717 non-null    object        
dtypes: datetime64[ns](1), int64(3), object(6)
memory usage: 56.1+ KB


In [44]:
df_victimas = df_victimas[["ID_hecho", "ROL", "SEXO", "EDAD", "FECHA_FALLECIMIENTO"]]
df_victimas = df_victimas.rename(columns= {"ID_hecho":"ID", "ROL":"Rol_vic", "SEXO":"Sexo", "EDAD": "Edad", "FECHA_FALLECIMIENTO": "Fecha_fallec" })
df_victimas.head(2)

ID    Rol_vic       Sexo Edad         Fecha_fallec
0  2016-0001  CONDUCTOR  MASCULINO   19  2016-01-01 00:00:00
1  2016-0002  CONDUCTOR  MASCULINO   70  2016-01-02 00:00:00

In [45]:
df_victimas["Sexo"] = df_victimas["Sexo"].replace({"MASCULINO":"M", "FEMENINO":"F"})

In [46]:
# Intenta convertir la columna a fechas
df_victimas["Fecha_fallec"] = pd.to_datetime(df_victimas["Fecha_fallec"], errors = "coerce")
df_victimas.head()

ID    Rol_vic Sexo Edad Fecha_fallec
0  2016-0001  CONDUCTOR    M   19   2016-01-01
1  2016-0002  CONDUCTOR    M   70   2016-01-02
2  2016-0003  CONDUCTOR    M   30   2016-01-03
3  2016-0004  CONDUCTOR    M   18          NaT
4  2016-0005  CONDUCTOR    M   29   2016-02-01

In [47]:
df_victimas["Fecha_fallec"] = df_victimas["Fecha_fallec"].apply(lambda x: "SI" if not pd.isnull(x) else "SD")
df_victimas

ID               Rol_vic Sexo Edad Fecha_fallec
0    2016-0001             CONDUCTOR    M   19           SI
1    2016-0002             CONDUCTOR    M   70           SI
2    2016-0003             CONDUCTOR    M   30           SI
3    2016-0004             CONDUCTOR    M   18           SD
4    2016-0005             CONDUCTOR    M   29           SI
..         ...                   ...  ...  ...          ...
712  2021-0092                PEATON    F   50           SI
713  2021-0093  PASAJERO_ACOMPAÑANTE    F   18           SI
714  2021-0094  PASAJERO_ACOMPAÑANTE    F   43           SI
715  2021-0095             CONDUCTOR    M   27           SI
716  2021-0096             CONDUCTOR    M   60           SI

[717 rows x 5 columns]

In [48]:
# Merge basado en la columna común "ID"
df = pd.merge(df_homicidios, df_victimas, on="ID")
df.head()

ID   Año  Mes Hora       Dia     Lugar        Calle_ppal  \
0  2016-0001  2016    1    4    Friday   AVENIDA  PIEDRA BUENA AV.   
1  2016-0002  2016    1    1  Saturday  GRAL PAZ    PAZ, GRAL. AV.   
2  2016-0003  2016    1    7    Sunday   AVENIDA    ENTRE RIOS AV.   
3  2016-0004  2016    1    0    Sunday   AVENIDA    LARRAZABAL AV.   
4  2016-0005  2016    1    5  Thursday   AVENIDA      SAN JUAN AV.   

                             Calle_sec  Comuna      Longitud       Latitud  \
0  FERNANDEZ DE LA CRUZ, F., GRAL. AV.       8  -58.47533969  -34.68757022   
1                  DE LOS CORRALES AV.       9  -58.50877521  -34.66977709   
2                                  NaN       1  -58.39040293  -34.63189362   
3             VILLEGAS, CONRADO, GRAL.       8  -58.46503904  -34.68092974   
4              SAENZ PE?A, LUIS, PRES.       1  -58.38718297  -34.62246630   

  Victima    Acusado    Rol_vic Sexo Edad Fecha_fallec  
0    MOTO       AUTO  CONDUCTOR    M   19           SI  
1    AUTO  PASAJEROS  CONDUCTOR    M   70           SI  
2    MOTO       AUTO  CONDUCTOR    M   30           SI  
3    MOTO         SD  CONDUCTOR    M   18           SD  
4    MOTO  PASAJEROS  CONDUCTOR    M   29           SI

In [49]:
df_comunas = pd.read_excel("../PI-2/01 - Datasets/Datasets Siniestros Viales/comunas.xlsx")
df_comunas = df_comunas[["COMUNAS", "BARRIOS"]]
df_comunas = df_comunas.rename(columns = {"COMUNAS":"Comuna", "BARRIOS": "Barrios"})
df_comunas.head()

Comuna                                  Barrios
0       2                                 RECOLETA
1       5                          ALMAGRO - BOEDO
2       6                                CABALLITO
3       7                FLORES - PARQUE CHACABUCO
4       9  LINIERS - MATADEROS - PARQUE AVELLANEDA

In [50]:
df_completo = pd.merge(df, df_comunas, on="Comuna")
df_completo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 715 entries, 0 to 714
Data columns (total 18 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   ID            715 non-null    object
 1   Año           715 non-null    int64 
 2   Mes           715 non-null    int64 
 3   Hora          715 non-null    object
 4   Dia           715 non-null    object
 5   Lugar         715 non-null    object
 6   Calle_ppal    715 non-null    object
 7   Calle_sec     540 non-null    object
 8   Comuna        715 non-null    int64 
 9   Longitud      715 non-null    object
 10  Latitud       715 non-null    object
 11  Victima       715 non-null    object
 12  Acusado       715 non-null    object
 13  Rol_vic       715 non-null    object
 14  Sexo          715 non-null    object
 15  Edad          715 non-null    object
 16  Fecha_fallec  715 non-null    object
 17  Barrios       715 non-null    object
dtypes: int64(3), object(15)
memory usage: 100.7+ KB


In [51]:
df_completo.to_csv("../PI-2/Datos_Homicidios.csv", index=False)

ARCHIVO "LESIONES.XLSX"

In [53]:
df = pd.read_excel("../PI-2/01 - Datasets/Datasets Siniestros Viales/lesiones.xlsx", sheet_name= "HECHOS")
df.head(2)

id  n_victimas  aaaa  mm  dd                fecha      hora  \
0  LC-2019-0000179           1  2019   1   1  2019-01-01 00:00:00  09:00:00   
1  LC-2019-0000053           1  2019   1   1  2019-01-01 00:00:00  01:55:00   

  franja_hora direccion_normalizada comuna  ...     latutid   victima acusado  \
0           9                    SD     14  ...  -34.559658  CICLISTA      SD   
1           1                    SD      8  ...  -34.669125      AUTO      SD   

   participantes moto auto transporte_publico camion ciclista gravedad  
0    CICLISTA-SD   SD   SD                 SD     SD        x       SD  
1        AUTO-SD   SD    x                 SD     SD       SD       SD  

[2 rows x 27 columns]

In [54]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23785 entries, 0 to 23784
Data columns (total 27 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   id                     23785 non-null  object 
 1   n_victimas             23785 non-null  int64  
 2   aaaa                   23785 non-null  int64  
 3   mm                     23785 non-null  int64  
 4   dd                     23785 non-null  int64  
 5   fecha                  23785 non-null  object 
 6   hora                   23785 non-null  object 
 7   franja_hora            23780 non-null  object 
 8   direccion_normalizada  23732 non-null  object 
 9   comuna                 23616 non-null  object 
 10  tipo_calle             23785 non-null  object 
 11  otra_direccion         23785 non-null  object 
 12  calle                  12867 non-null  object 
 13  altura                 12771 non-null  float64
 14  cruce                  9407 non-null   object 
 15  ge

In [55]:
df = df[["id", "franja_hora", "comuna", "tipo_calle", "acusado"]]
df.head(2)

id franja_hora comuna tipo_calle acusado
0  LC-2019-0000179           9     14         SD      SD
1  LC-2019-0000053           1      8         SD      SD

In [57]:
df_hechos = df.rename(columns={"id":"ID", "franja_hora":"Hora", "comuna":"Comuna", "tipo_calle":"Lugar", "acusado":"Acusado"})
df_hechos.head(2)

ID Hora Comuna Lugar Acusado
0  LC-2019-0000179    9     14    SD      SD
1  LC-2019-0000053    1      8    SD      SD

In [58]:
df = pd.read_excel("../PI-2/01 - Datasets/Datasets Siniestros Viales/lesiones.xlsx", sheet_name= "VICTIMAS")
df.head(2)

ID hecho   AAA  MM  DD     FECHA  VEHICULO_VICTIMA   SEXO  \
0  LC-2019-0000053  2019   1   1 2019-01-01               sd  Varon   
1  LC-2019-0000063  2019   1   1 2019-01-01               sd     SD   

  EDAD_VICTIMA GRAVEDAD  
0           57       SD  
1           SD       SD

In [59]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27605 entries, 0 to 27604
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   ID hecho          27605 non-null  object        
 1   AAA               27605 non-null  int64         
 2   MM                27605 non-null  int64         
 3   DD                27605 non-null  int64         
 4   FECHA             27605 non-null  datetime64[ns]
 5   VEHICULO_VICTIMA  27605 non-null  object        
 6   SEXO              27605 non-null  object        
 7   EDAD_VICTIMA      27605 non-null  object        
 8   GRAVEDAD          27605 non-null  object        
dtypes: datetime64[ns](1), int64(3), object(5)
memory usage: 1.9+ MB


In [60]:
df = df[["ID hecho", "AAA", "MM", "VEHICULO_VICTIMA", "SEXO", "EDAD_VICTIMA", "GRAVEDAD"]]
df.head(2)

ID hecho   AAA  MM VEHICULO_VICTIMA   SEXO EDAD_VICTIMA GRAVEDAD
0  LC-2019-0000053  2019   1               sd  Varon           57       SD
1  LC-2019-0000063  2019   1               sd     SD           SD       SD

In [61]:
df_victimas = df.rename(columns={"ID hecho":"ID", "AAA":"Año", "MM":"Mes", "VEHICULO_VICTIMA":"Victima", "SEXO":"Sexo", "EDAD_VICTIMA": "Edad", "GRAVEDAD": "Gravedad"})
df_victimas.head(2)

ID   Año  Mes Victima   Sexo Edad Gravedad
0  LC-2019-0000053  2019    1      sd  Varon   57       SD
1  LC-2019-0000063  2019    1      sd     SD   SD       SD

In [62]:
df_victimas["Gravedad"].unique()

array(['SD', 'sd', 'GRAVE', 'grave'], dtype=object)

In [64]:
df_victimas["Gravedad"] = df_victimas["Gravedad"].replace({"sd":"LEVE", "SD": "LEVE", "grave":"GRAVE"})
df_victimas["Gravedad"].unique()

array(['LEVE', 'GRAVE'], dtype=object)

In [65]:
# Merge basado en la columna común "ID"
df_lesiones = pd.merge(df_victimas, df_hechos, on="ID")
df_lesiones.head()

ID   Año  Mes Victima   Sexo Edad Gravedad Hora Comuna Lugar  \
0  LC-2019-0000053  2019    1      sd  Varon   57     LEVE    1      8    SD   
1  LC-2019-0000063  2019    1      sd     SD   SD     LEVE    2      8    SD   
2  LC-2019-0000079  2019    1      sd  Varon   SD     LEVE    2      7    SD   
3  LC-2019-0000082  2019    1      sd  Varon   45     LEVE    4      3    SD   
4  LC-2019-0000082  2019    1      sd  Mujer   45     LEVE    4      3    SD   

  Acusado  
0      SD  
1      SD  
2      SD  
3      SD  
4      SD

In [66]:
df_lesiones.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27598 entries, 0 to 27597
Data columns (total 11 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   ID        27598 non-null  object
 1   Año       27598 non-null  int64 
 2   Mes       27598 non-null  int64 
 3   Victima   27598 non-null  object
 4   Sexo      27598 non-null  object
 5   Edad      27598 non-null  object
 6   Gravedad  27598 non-null  object
 7   Hora      27593 non-null  object
 8   Comuna    27420 non-null  object
 9   Lugar     27598 non-null  object
 10  Acusado   27598 non-null  object
dtypes: int64(2), object(9)
memory usage: 2.3+ MB


In [67]:
df_lesiones.to_csv("../PI-2/Datos_Lesiones.csv", index=False)